@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -552,14 +586,27 @@ export class VisibleTileSet {
                 if (!tile.allGeometryLoaded) {
                     numTilesLoading++;
                 } else {
-                    tile.numFramesVisible++;
                     // If this tile's data source is "covering" then other tiles beneath it have
                     // their rendering skipped, see [[Tile.willRender]].
                     this.skipOverlappedTiles(dataSource, tile);
 
                     if (tile.frameNumVisible < 0) {
-                        // Store the fist frame the tile became visible.
-                        tile.frameNumVisible = frameNumber;
+                        newTilesPerFrame++;
+                        if (
+                            // if set to 0, it will ignore the limit and upload all available
+                            this.options.maxTilesPerFrame !== 0 &&
+                            newTilesPerFrame > this.options.maxTilesPerFrame &&
+                            //if the tile was already visible last frame dont delay it
+                            !(tile.frameNumLastVisible === dataSource.mapView.frameNumber - 1)
+                        ) {
+                            tile.delayRendering = true;
+                            tile.mapView.update();
+                        } else {
+                            tile.numFramesVisible++;
```

Comment:
this will only ever be 1, because this code is only executed if `framNumVisible` is < 0. Should be fixed.

Reply:
right, good catch, fixed it

Reply:
right, good catch. fixed it